### PR TITLE
Remember package refactor17388

### LINKED ISSUE #17388: Remembering application when moving a method to another package
When we move a method as extension of a package, the system should remember the selected package so that the second time we do not have to redo it. 
pour le moment on a  
Capture 2024-11-17 at 11 11 14
Capture 2024-11-17 at 11 12 24

puis encore 
Capture 2024-11-17 at 11 11 14

### PR DESCRIPTION
**Pull request by AlexisCnockaert**
Fixes: #17388 
I created a class variable to remember lastselected package, the next time its searched by default

Screen Shot 2024-11-19 at 11 01 02

Here it is remembered : 

Screen Shot 2024-11-19 at 11 01 11

### CODE REVIEW

File: src/Calypso-SystemTools-Core/ClySystemBrowserContext.class.st [hunk 1]
```diff
@@ -45,11 +45,26 @@ And I implement simplified search requests:
...
 	#superclass : 'ClyBrowserContext',
+	#classInstVars : [
+		'RememberedPackage'
+	],
 	#category : 'Calypso-SystemTools-Core-Contexts',
 	#package : 'Calypso-SystemTools-Core',
 	#tag : 'Contexts'
 }
 
+{ #category : 'as yet unclassified' }
```
Comment (jecisc) [inline src/Calypso-SystemTools-Core/ClySystemBrowserContext.class.st]: The method needs to be classified

File: src/Calypso-SystemTools-Core/ClySystemBrowserContext.class.st [hunk 2]
```diff
@@ -45,11 +45,26 @@ And I implement simplified search requests:
...
+ClySystemBrowserContext class >> rememberedPackage [
+
+	^ RememberedPackage ifNil: [ RememberedPackage := nil ]
+]
+
+{ #category : 'as yet unclassified' }
+ClySystemBrowserContext class >> rememberedPackage: aPackage [
+
+RememberedPackage := aPackage
```
Comment (jecisc) [inline src/Calypso-SystemTools-Core/ClySystemBrowserContext.class.st]: I recommend you to format your methods before compiling (Cmd + shift + f)

**Comment by Ducasse** [comment]
Pay attention 
Instance variable names must start with a lowercase letter: {ClySystemBrowserContext class}

**Comment by Ducasse** [comment]
Alexis you should use a class variable (sharedVariables: and not an instance variables of the metaclass. Please read the Pharo by example chapter so that you get it right because knowing the difference it important. 
There is also a video on the pharo mooc.

**Comment by AlexisCnockaert** [comment]
> Alexis you should use a class variable (sharedVariables: and not an instance variables of the metaclass. Please read the Pharo by example chapter so that you get it right because knowing the difference it important.
> There is also a video on the pharo mooc.

Sorry I'll pay attention next time

**Pull request merged**